During the incremental migration of PostgreSQL, if the update statement is executed, the `Sharding value must implements Comparable` exception will occur

## Bug Report

### Which version of ShardingSphere did you use?
Master f2151421374e3e5a856f82a7283de4d3bb0ad154
### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy
### Expected behavior
PostgreSQL successfully migrated during data migration of multiple data sources
### Actual behavior
During the incremental migration of PostgreSQL, if the update statement is executed, the `Sharding value must implements Comparable` exception will occur
### Reason analyze (If you can)
`oldValue` in ```org.apache.shardingsphere.data.pipeline.api.ingest.record.Column``` cause
### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
1. Use shardingsphere-proxy to add resources
2. Create migration task
3. Use update statement during incremental migration
4. 
```java
Caused by: java.lang.IllegalArgumentException: Sharding value must implements Comparable.
    at com.google.common.base.Preconditions.checkArgument(Preconditions.java:142)
    at org.apache.shardingsphere.sharding.route.engine.condition.generator.ConditionValue.getValue(ConditionValue.java:53)
    at org.apache.shardingsphere.sharding.route.engine.condition.generator.ConditionValue.getValue(ConditionValue.java:41)
    at org.apache.shardingsphere.sharding.route.engine.condition.generator.ConditionValue.<init>(ConditionValue.java:36)
    at org.apache.shardingsphere.sharding.route.engine.condition.generator.impl.ConditionValueCompareOperatorGenerator.generate(ConditionValueCompareOperatorGenerator.java:64)
    at org.apache.shardingsphere.sharding.route.engine.condition.generator.ConditionValueGeneratorFactory.generate(ConditionValueGeneratorFactory.java:51)
    at org.apache.shardingsphere.sharding.route.engine.condition.engine.impl.WhereClauseShardingConditionEngine.createShardingConditionValueMap(WhereClauseShardingConditionEngine.java:142)
    at org.apache.shardingsphere.sharding.route.engine.condition.engine.impl.WhereClauseShardingConditionEngine.createShardingConditions(WhereClauseShardingConditionEngine.java:84)
    at org.apache.shardingsphere.sharding.route.engine.condition.engine.impl.WhereClauseShardingConditionEngine.createShardingConditions(WhereClauseShardingConditionEngine.java:74)
    at org.apache.shardingsphere.sharding.route.engine.ShardingSQLRouter.createShardingConditions(ShardingSQLRouter.java:66)
    at org.apache.shardingsphere.sharding.route.engine.ShardingSQLRouter.createRouteContext(ShardingSQLRouter.java:50)
    at org.apache.shardingsphere.sharding.route.engine.ShardingSQLRouter.createRouteContext(ShardingSQLRouter.java:44)
    at org.apache.shardingsphere.infra.route.engine.impl.PartialSQLRouteExecutor.route(PartialSQLRouteExecutor.java:73)
    at org.apache.shardingsphere.infra.route.engine.SQLRouteEngine.route(SQLRouteEngine.java:53)
    at org.apache.shardingsphere.infra.context.kernel.KernelProcessor.route(KernelProcessor.java:54)
    at org.apache.shardingsphere.infra.context.kernel.KernelProcessor.generateExecutionContext(KernelProcessor.java:46)
    at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSpherePreparedStatement.createExecutionContext(ShardingSpherePreparedStatement.java:381)
    at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSpherePreparedStatement.execute(ShardingSpherePreparedStatement.java:290)
    at org.apache.shardingsphere.data.pipeline.core.importer.AbstractImporter.executeUpdate(AbstractImporter.java:201)
    at org.apache.shardingsphere.data.pipeline.core.importer.AbstractImporter.executeUpdate(AbstractImporter.java:185)
    at org.apache.shardingsphere.data.pipeline.core.importer.AbstractImporter.doFlush(AbstractImporter.java:157)
    at org.apache.shardingsphere.data.pipeline.core.importer.AbstractImporter.tryFlush(AbstractImporter.java:139)
    at org.apache.shardingsphere.data.pipeline.core.importer.AbstractImporter.flushInternal(AbstractImporter.java:130)
    at org.apache.shardingsphere.data.pipeline.core.importer.AbstractImporter.lambda$flush$2(AbstractImporter.java:124)
    at java.util.ArrayList.forEach(ArrayList.java:1257)
    at org.apache.shardingsphere.data.pipeline.core.importer.AbstractImporter.flush(AbstractImporter.java:116)
    at org.apache.shardingsphere.data.pipeline.core.importer.AbstractImporter.write(AbstractImporter.java:97)
    at org.apache.shardingsphere.data.pipeline.core.importer.AbstractImporter.start(AbstractImporter.java:86)
    ... 8 common frames omitted
```
### Example codes for reproduce this issue (such as a github link).

--- a/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/OpenGaussWalDumper.java
+++ b/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/OpenGaussWalDumper.java
@@ -24,11 +24,8 @@ import org.apache.shardingsphere.data.pipeline.api.datasource.config.impl.Standa
 import org.apache.shardingsphere.data.pipeline.api.executor.AbstractLifecycleExecutor;
 import org.apache.shardingsphere.data.pipeline.api.ingest.channel.Channel;
 import org.apache.shardingsphere.data.pipeline.api.ingest.position.IngestPosition;
-import org.apache.shardingsphere.data.pipeline.api.ingest.record.Column;
-import org.apache.shardingsphere.data.pipeline.api.ingest.record.DataRecord;
 import org.apache.shardingsphere.data.pipeline.api.ingest.record.Record;
 import org.apache.shardingsphere.data.pipeline.core.datasource.creator.PipelineDataSourceCreatorFactory;
-import org.apache.shardingsphere.data.pipeline.core.ingest.IngestDataChangeType;
 import org.apache.shardingsphere.data.pipeline.core.ingest.exception.IngestException;
 import org.apache.shardingsphere.data.pipeline.core.util.ThreadUtil;
 import org.apache.shardingsphere.data.pipeline.opengauss.ingest.wal.OpenGaussLogicalReplication;
@@ -122,7 +119,6 @@ public final class OpenGaussWalDumper extends AbstractLifecycleExecutor implemen
                 if (!(event instanceof PlaceholderEvent) && log.isDebugEnabled()) {
                     log.debug("dump, event={}, record={}", event, record);
                 }
-                updateRecordOldValue(record);
                 pushRecord(record);
             }
         } catch (final SQLException ex) {
@@ -130,21 +126,6 @@ public final class OpenGaussWalDumper extends AbstractLifecycleExecutor implemen
                 return;
             }
             throw new IngestException(ex);
-        }
-    }
-    
-    private void updateRecordOldValue(final Record record) {
-        if (!(record instanceof DataRecord)) {
-            return;
-        }
-        DataRecord dataRecord = (DataRecord) record;
-        if (!IngestDataChangeType.UPDATE.equals(dataRecord.getType())) {
-            return;
-        }
-        for (Column each: dataRecord.getColumns()) {
-            if (each.isPrimaryKey() && each.isUpdated()) {
-                each.setOldValue(each.getValue());
-            }
         }
     }
     

--- a/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-postgresql/src/main/java/org/apache/shardingsphere/data/pipeline/postgresql/ingest/wal/WalEventConverter.java
+++ b/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-postgresql/src/main/java/org/apache/shardingsphere/data/pipeline/postgresql/ingest/wal/WalEventConverter.java
@@ -126,9 +126,7 @@ public final class WalEventConverter {
     private void putColumnsIntoDataRecord(final DataRecord dataRecord, final TableMetaData tableMetaData, final List<Object> values) {
         for (int i = 0; i < values.size(); i++) {
             Object primaryKeyOldValue = tableMetaData.isPrimaryKey(i) ? values.get(i) : null;
-            Column column = new Column(tableMetaData.getColumnMetaData(i).getName(),
-                    primaryKeyOldValue,
-                    values.get(i), true, tableMetaData.isPrimaryKey(i));
+            Column column = new Column(tableMetaData.getColumnMetaData(i).getName(), primaryKeyOldValue, values.get(i), true, tableMetaData.isPrimaryKey(i));
             dataRecord.addColumn(column);
         }
     }

--- a/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-postgresql/src/main/java/org/apache/shardingsphere/data/pipeline/postgresql/ingest/wal/WalEventConverter.java
+++ b/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-postgresql/src/main/java/org/apache/shardingsphere/data/pipeline/postgresql/ingest/wal/WalEventConverter.java
@@ -125,7 +125,11 @@ public final class WalEventConverter {
     
     private void putColumnsIntoDataRecord(final DataRecord dataRecord, final TableMetaData tableMetaData, final List<Object> values) {
         for (int i = 0; i < values.size(); i++) {
-            dataRecord.addColumn(new Column(tableMetaData.getColumnMetaData(i).getName(), values.get(i), true, tableMetaData.isPrimaryKey(i)));
+            Object primaryKeyOldValue = tableMetaData.isPrimaryKey(i) ? values.get(i) : null;
+            Column column = new Column(tableMetaData.getColumnMetaData(i).getName(),
+                    primaryKeyOldValue,
+                    values.get(i), true, tableMetaData.isPrimaryKey(i));
+            dataRecord.addColumn(column);
         }
     }
 }

--- a/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/main/java/org/apache/shardingsphere/data/pipeline/api/ingest/record/Column.java
+++ b/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/main/java/org/apache/shardingsphere/data/pipeline/api/ingest/record/Column.java
@@ -18,11 +18,12 @@
 package org.apache.shardingsphere.data.pipeline.api.ingest.record;
 
 import lombok.Getter;
-import lombok.Setter;
+import lombok.RequiredArgsConstructor;
 
 /**
  * Column.
  */
+@RequiredArgsConstructor
 @Getter
 public final class Column {
     
@@ -31,8 +32,7 @@ public final class Column {
     /**
      * Value are available only when the primary key column is updated.
      */
-    @Setter
-    private Object oldValue;
+    private final Object oldValue;
     
     private final Object value;
     
@@ -42,14 +42,6 @@ public final class Column {
     
     public Column(final String name, final Object value, final boolean updated, final boolean primaryKey) {
         this(name, null, value, updated, primaryKey);
-    }
-    
-    public Column(final String name, final Object oldValue, final Object value, final boolean updated, final boolean primaryKey) {
-        this.name = name;
-        this.oldValue = oldValue;
-        this.value = value;
-        this.updated = updated;
-        this.primaryKey = primaryKey;
     }
     
     @Override

